### PR TITLE
optionally render ion-icon from asset path if provided

### PR DESCRIPTION
**Pull request by jfspencer**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features) (_didn't see a spec file, happy to add tests, just need be pointed in the right direction_)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: #18968 

## What is the new behavior?
supplying `iconSrc` on a ToastButton config object will use <ion-icon src="iconSrc">

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
- Running `npm run build` updated the vue package-lock file. I can remove that from the PR if needed.

### CODE REVIEW

**Comment by jfspencer** [comment]
looks like the failure is being caused by a chrome update (to 76)

File: core/src/components/toast/toast.tsx [hunk 1]
```diff
@@ -241,12 +241,7 @@ export class Toast implements ComponentInterface, OverlayInterface {
         {buttons.map(b =>
           <button type="button" class={buttonClass(b)} tabIndex={0} onClick={() => this.buttonClick(b)}>
             <div class="toast-button-inner">
-              {b.icon &&
-                <ion-icon
-                  name={b.icon}
```
Comment (manucorporat) [inline core/src/components/toast/toast.tsx]: If you change `name={i.con}` we can accept icon name and path all at once

**Comment by manucorporat** [comment]
I like the feature, but we probably don't need a new iconSrc API, read my comment ^

**Comment by jfspencer** [comment]
@manucorporat @brandyscarney thanks, updated the PR accordingly.

**Comment by manucorporat** [comment]
Awesome! this looks great, last two things, 
can you revert the change in vue/package-lock.json ?
also make sure you run `npm run build` in core and commit the changes!

**Comment by jfspencer** [comment]
@manucorporat 
- [x] remove vue/package-lock
- [x] commit generated changes after `npm run build` on core

**Comment by manucorporat** [comment]
@jfspencer beautiful!
will try to get it merged before the next release!

**Pull request merged**